client/webserver/site/src/js/forms.js

```diff
@@ -298,31 +299,70 @@ export class WalletConfigForm {
  * ConfirmRegistrationForm should be used with the "confirmRegistrationForm" template.
  */
 export class ConfirmRegistrationForm {
-  constructor (application, form, { getDexAddr, getCertFile }, success, insufficientFundsFail) {
+  constructor (application, form, { getDexAddr, getCertFile }, success, insufficientFundsFail, setupWalletFn) {
     this.fields = Doc.parsePage(form, [
       'feeDisplay', 'marketRowTemplate', 'marketsTableRows', 'appPass', 'appPassBox',
-      'appPassSpan', 'submitConfirm', 'regErr'
+      'appPassSpan', 'submitConfirm', 'regErr', 'feeRowTemplate', 'feeTableRows',
+      'setupWallet'
     ])
     app = application
     this.getDexAddr = getDexAddr
     this.getCertFile = getCertFile
     this.success = success
     this.insufficientFundsFail = insufficientFundsFail
     this.form = form
-    bind(form, this.fields.submitConfirm, () => this.submitForm())
+    this.setupWalletFn = setupWalletFn
+    this.feeWallet = null
+    bind(form, this.fields.submitConfirm, () => this.submitForm(this.feeWallet))
+  }
+
+  setWalletFee (assetId) {
+    this.feeWallet = assetId
   }
 
   /*
    * setExchange populates the form with the details of an exchange.
    */
   setExchange (xc) {
+    // store xc in case we need to refresh the data, like after setupping
+    // a new wallet.
+    this.xc = xc
     const fields = this.fields
     this.fees = xc.regFees
     const dcr = this.fees.dcr // TODO: display all options and give choice
     fields.feeDisplay.textContent = `${Doc.formatCoinValue(dcr.amount / 1e8)} ${app.assets[dcr.id].symbol.toUpperCase()}`
     while (fields.marketsTableRows.firstChild) {
       fields.marketsTableRows.removeChild(fields.marketsTableRows.firstChild)
     }
+    while (fields.feeTableRows.firstChild) {
+      fields.feeTableRows.removeChild(fields.feeTableRows.firstChild)
+    }
+    const feeAsset = Object.values(xc.regFees)
+    feeAsset.forEach((fee) => {
+      const tr = fields.feeRowTemplate.cloneNode(true)
+      const asset = Doc.assetById(fee.id)
+      Doc.tmplElement(tr, 'asseticon').src = Doc.logoPath(asset)
+      Doc.tmplElement(tr, 'asset').innerText = asset.toUpperCase()
+      Doc.tmplElement(tr, 'confs').innerText = fee.confs
+      Doc.tmplElement(tr, 'fee').innerText = fee.amount / 1e8
+
+      const haveWallet = app.user.assets[fee.id] ? app.user.assets[fee.id].wallet : null
+      const setupWallet = tr.querySelector('#setupWallet')
+      if (haveWallet) {
+        setupWallet.innerText = intl.prep(intl.ID_CHOOSE_WALLET)
+        Doc.bind(setupWallet, 'click', () => this.setWalletFee(fee.id))
```

Comment:
I think this action should also update a something to indicate the selection was made and that clicking "Register" will in fact use the intended asset and amount.

For this I had envisioned either or both of: (1) modify the table row's style or just the clicked this button to show it is selected (sorry if this is already done, I haven't tested), which would obviously require undoing this styling change if the user clicks another row, or (2) Update a new sentence just above the Password field to say something like "This will pay a fee of [X amount] [Y asset symbol]."

Also, the styling of the button would be different (perhaps color) between Choose and Setup

Reply:
I'm pretty happy now with how solution (1) is implemented.  With a boundary around the selected asset's row, it's quite clear what is going to happen when you submit the form.

Maybe another reviewer will think the text proposed in (2) is also valuable, but I can live without it.